### ISSUE TITLE
DaemonSet "ebs-csi-node" is invalid

### ISSUE THREAD

**Issue by caseybrown89** [kubernetes-sigs/aws-ebs-csi-driver#1347]
/kind bug

**What happened?**

Running through https://aws.amazon.com/premiumsupport/knowledge-center/eks-persistent-storage/ to setup the Amazon EBS CSI driver. I ran into an issue executing the following:

>7.    To deploy the Amazon EBS CSI driver, run one of the following commands based on your Region:
>All Regions other than China Regions:
> `kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=master"`

I get the following error:

```
kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=master"
serviceaccount/ebs-csi-controller-sa created
serviceaccount/ebs-csi-node-sa created
clusterrole.rbac.authorization.k8s.io/ebs-csi-node-role created
clusterrole.rbac.authorization.k8s.io/ebs-external-attacher-role created
clusterrole.rbac.authorization.k8s.io/ebs-external-provisioner-role created
clusterrole.rbac.authorization.k8s.io/ebs-external-resizer-role created
clusterrole.rbac.authorization.k8s.io/ebs-external-snapshotter-role created
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-attacher-binding created
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-node-getter-binding created
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-provisioner-binding created
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-resizer-binding created
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-snapshotter-binding created
deployment.apps/ebs-csi-controller created
poddisruptionbudget.policy/ebs-csi-controller created
csidriver.storage.k8s.io/ebs.csi.aws.com created
The DaemonSet "ebs-csi-node" is invalid: spec.template.spec.containers[0].securityContext: Invalid value: core.SecurityContext{Capabilities:(*core.Capabilities)(nil), Privileged:(*bool)(0xc006c6ec5a), SELinuxOptions:(*core.SELinuxOptions)(nil), WindowsOptions:(*core.WindowsSecurityContextOptions)(nil), RunAsUser:(*int64)(nil), RunAsGroup:(*int64)(nil), RunAsNonRoot:(*bool)(nil), ReadOnlyRootFilesystem:(*bool)(0xc006c6ec5b), AllowPrivilegeEscalation:(*bool)(0xc006c6ec3a), ProcMount:(*core.ProcMountType)(nil), SeccompProfile:(*core.SeccompProfile)(nil)}: cannot set `allowPrivilegeEscalation` to false and `privileged` to true
```

**What you expected to happen?**

I expect this step to succeed.

**How to reproduce it (as minimally and precisely as possible)?**

Follow the guide to Step 7, execute the step.

**Anything else we need to know?**:

This appears to be tied to a recent change in release [v1.11.0](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/releases/tag/v1.11.0). Running against v.1.10.0 works:

```
kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=v1.10.0"
serviceaccount/ebs-csi-controller-sa unchanged
serviceaccount/ebs-csi-node-sa unchanged
clusterrole.rbac.authorization.k8s.io/ebs-csi-node-role unchanged
clusterrole.rbac.authorization.k8s.io/ebs-external-attacher-role unchanged
clusterrole.rbac.authorization.k8s.io/ebs-external-provisioner-role unchanged
clusterrole.rbac.authorization.k8s.io/ebs-external-resizer-role unchanged
clusterrole.rbac.authorization.k8s.io/ebs-external-snapshotter-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-attacher-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-node-getter-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-provisioner-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-resizer-binding unchanged
clusterrolebinding.rbac.authorization.k8s.io/ebs-csi-snapshotter-binding unchanged
deployment.apps/ebs-csi-controller configured
Warning: policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
poddisruptionbudget.policy/ebs-csi-controller configured
daemonset.apps/ebs-csi-node created
csidriver.storage.k8s.io/ebs.csi.aws.com unchanged
```

**Environment**
- Kubernetes version (use `kubectl version`):

```
kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:30:46Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.11-eks-18ef993", GitCommit:"b9628d6d3867ffd84c704af0befd31c7451cdc37", GitTreeState:"clean", BuildDate:"2022-07-06T18:06:23Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.24) and server (1.22) exceeds the supported minor version skew of +/-1
```

I also testing with a matching client, no change
```
./kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.12", GitCommit:"b058e1760c79f46a834ba59bd7a3486ecf28237d", GitTreeState:"clean", BuildDate:"2022-07-13T14:59:18Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.11-eks-18ef993", GitCommit:"b9628d6d3867ffd84c704af0befd31c7451cdc37", GitTreeState:"clean", BuildDate:"2022-07-06T18:06:23Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```
- Driver version:
master, 1.11.0, 1.10.0

**Comment by torredil** [comment]
Release `v1.11.0` was aborted due to https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1346. Please use the latest version which is `v1.10.0`. We are currently working on releasing a new version, `v1.11.1` which will be announced in the [CHANGELOG](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/CHANGELOG.md) when complete.

**Comment by ConnorJC3** [comment]
@caseybrown89 major facepalm, it's because the fix from #1333 for this didn't make its way into the kustomize manifests. This somehow slipped through manual testing too. Opening a PR to fix this now.
